runner/full.go

```diff
@@ -67,6 +68,8 @@ func collectDiffAndSubmit(server *state.Server, globalCollectionOpts state.Colle
 		newState.StatementStats = transientState.ResetStatementStats
 	}
 
+	newState.QueryIdentities = pruneQueryIdentities(newState.QueryIdentities)
```

Comment:
As noted in a parallel comment, I think it may be better to put `QueryIdentities` directly on the `Server` struct, since it doesn't follow the usual prev state => new state handling, and is used outside of full snapshots.